src/base/dof_map_constraints.C

```diff
@@ -431,6 +431,9 @@ private:
         std::vector<bool> is_boundary_node(elem->n_nodes(), false),
           is_boundary_edge(elem->n_edges(), false),
           is_boundary_side(elem->n_sides(), false);
+
+        // We also maintain a separate list of nodeset-based boundary nodes
+        std::vector<bool> is_boundary_nodeset(elem->n_nodes(), false);
```

Comment:
Even if there's no chance this code will run threaded (is there?) I'd still like to lodge my usual formal complaint against `std::vector<bool>`.

See section on [data races](http://www.cplusplus.com/reference/vector/vector-bool/) at the bottom.